### PR TITLE
[Mobile] Changed the columns per row from 2 two 3

### PR DESCRIPTION
**Pull request by antonis**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2384

## Related PRs:
gutenberg-mobile https://github.com/wordpress-mobile/gutenberg-mobile/pull/2408
WordPress-iOS https://github.com/wordpress-mobile/WordPress-iOS/pull/14356

## Description
This fix changes the number of blocks per line from two to three.
When more than two block rows are rendered we need to change the justify content of the parent block-list component to align the blocks at the beginning of the row.

### Two columns
* Create a columns block with columns columns
* Make sure that you see two blocks in the row sharing the available space
### Three columns
* Create a columns block with more three columns
* Make sure that you see three blocks in the row
### Four columns
* Create a columns block with more four columns
* Make sure that you see three blocks in the first row and one in the second one aligned at the start
### Generic Test Case
* Create a columns block with more than two columns
* Make sure that you see three blocks three blocks per full row and the remaining blocks are aligned at the start of the last row

## Screenshots
Simulator Screen Shot - iPhone 11 - 2020-06-19 at 16 13 50

## Types of changes
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/columns/edit.native.js [hunk 1]
```diff
@@ -44,6 +44,13 @@ const ALLOWED_BLOCKS = [ 'core/column' ];
 const DEFAULT_COLUMNS = 2;
 const MIN_COLUMNS_NUMBER = 1;
 
+/**
+ * Number of columns in each row for large breakpoint container width
+ *
+ * @type {number}
+ */
+const LARGE_CONTAINER_COLUMNS_IN_ROW = 3;
```
Comment (lukewalczak) [inline packages/block-library/src/columns/edit.native.js:52]: Wdyt about renaming `LARGE_CONTAINER_COLUMNS_IN_ROW` to `MAX_COLUMNS_NUM_IN_ROW`? Then we can also rename:

`DEFAULT_COLUMNS` to `DEFAULT_COLUMNS_NUM` and `MIN_COLUMNS_NUMBER` to `MIN_COLUMNS_NUM` to keep the consistency ✌️

File: packages/block-library/src/columns/edit.native.js [hunk 2]
```diff
@@ -95,8 +105,11 @@ function ColumnsEditContainer( {
 			// show only 1 Column in row for mobile breakpoint container width
 			return 1;
 		} else if ( containerWidth < BREAKPOINTS.large ) {
-			// show 2 Column in row for large breakpoint container width
-			return Math.min( Math.max( 1, columnCount ), 2 );
+			// show LARGE_CONTAINER_COLUMNS_IN_ROW Column in row for large breakpoint container width
```
Comment (lukewalczak) [inline packages/block-library/src/columns/edit.native.js:108]: Small nit: this comment actually is falsy, because when adding fresh Columns block it will show 2 columns instead of `LARGE_CONTAINER_COLUMNS_IN_ROW` for the large breakpoint.

Maybe rephrase it to: show the maximum number of columns in a row (up to max 3) for large breakpoint container width.

File: packages/block-library/src/columns/edit.native.js (hunk 1 above)
Comment (antonis) [reply packages/block-library/src/columns/edit.native.js]: Hello @lucasstark 👋 
Your proposal sounds good. I submitted the changes.

Comment (lukewalczak) [reply packages/block-library/src/columns/edit.native.js]: Thanks!

**Review by lukewalczak — APPROVED** [review]

**Review by chipsnyder — APPROVED** [review]
This LGTM! Nice work @antonis 

Since we changed the margins slightly between 2 and 3 columns block I would like to see what @iamthomasbishop says in https://github.com/wordpress-mobile/WordPress-iOS/pull/14356#issuecomment-650230495 also would be curious to know if the left aligned is the desired outcome or what we should do there.

**Comment by chipsnyder** [comment]
Toggling PR as it looks like some CI jobs weren't associated to this PR.
